staging/src/k8s.io/csi-translation-lib/translate_test.go

```diff
@@ -77,6 +89,201 @@ func TestTranslationStability(t *testing.T) {
 	}
 }
 
+func TestTopologyTranslation(t *testing.T) {
+	testCases := []struct {
+		name             string
+		pv               *v1.PersistentVolume
+		expectedTopology []*v1.NodeSelectorRequirement
+	}{
+		{
+			name: "GCE PD with zone labels",
+			pv:   newGCEPDPV(defaultZoneLabels, nil /*topology*/),
+			expectedTopology: []*v1.NodeSelectorRequirement{
+				newTopology(plugins.GCEPDTopologyKey, "us-east-1"),
+			},
+		},
+		{
+			name: "GCE PD with existing topology",
+			pv:   newGCEPDPV(nil /*labels*/, newTopology(v1.LabelZoneFailureDomain, "us-east-2")),
+			expectedTopology: []*v1.NodeSelectorRequirement{
+				newTopology(v1.LabelZoneFailureDomain, "us-east-2"),
+				newTopology(plugins.GCEPDTopologyKey, "us-east-2"),
```

Comment:
We should keep the in-tree topology, correct?

Reply:
im not sure this makes sense to me - will observers/consumers try to use both of these clauses and if for some reason it doesn't understand one or the other it wont be able to match to this requirement?

Reply:
That's what I'd expect, but I haven't looked at how multiple requirements are handled.

I wasn't sure how to handle this scenario (existing topology with beta label). From comment https://github.com/kubernetes/kubernetes/pull/83394#discussion_r341396679 I learned we have to handle it, but I wasn't sure if I should *replace* the old topology or simple *add* the new one.

Reply:
/cc @msau42

Reply:
I think we should replace the old one.  Having them together makes them an "AND", and the beta labels are being removed, so the AND will fail.